### PR TITLE
feat(backup): recording recurring jobs test case

### PR DESCRIPTION
**Pull request by mantissahz**
Add the test case for backing up and restoring recurring jobs and groups.

longhorn/longhorn#2227

### CODE REVIEW

File: manager/integration/tests/common.py [hunk 1]
```diff
@@ -181,6 +181,7 @@
 SETTING_CONCURRENT_AUTO_ENGINE_UPGRADE_NODE_LIMIT = \
     "concurrent-automatic-engine-upgrade-per-node-limit"
 SETTING_SUPPORT_BUNDLE_FAILED_LIMIT = "support-bundle-failed-history-limit"
+SETTING_RESTORE_RECURRINGJOBS = "restore-volume-recurring-jobs"
```
Comment (c3y1huang) [inline manager/integration/tests/common.py:184]:
```suggestion
SETTING_RESTORE_RECURRING_JOBS = "restore-volume-recurring-jobs"
```

**Review by c3y1huang — CHANGES_REQUESTED** [review]

File: manager/integration/tests/test_recurring_job.py [hunk 2]
```diff
@@ -1687,3 +1690,121 @@ def test_recurring_job_backup(set_random_backupstore, client, batch_v1_api):  #
...
+    complete_backup_1_count = 0
+    restore_snapshot_name = ""
+    volume = client.by_id_volume(volume_name1)
+    wait_for_backup_completion(client, volume_name1)
+    for b in volume.backupStatus:
+        if back1+"-" in b.snapshot:
+            complete_backup_1_count += 1
+            restore_snapshot_name = b.snapshot
+
+    # 1 completed backups from backup1
```
Comment (c3y1huang) [inline manager/integration/tests/test_recurring_job.py:1787]: This comment seems redundant.

File: manager/integration/tests/common.py (hunk 1 above)
Comment (mantissahz) [reply manager/integration/tests/common.py]: Modified. Thanks.

File: manager/integration/tests/test_recurring_job.py (hunk 2 above)
Comment (mantissahz) [reply manager/integration/tests/test_recurring_job.py]: Removed. Thanks.

**Review by c3y1huang — APPROVED** [review]

**Review by innobead — APPROVED** [review]

**Pull request merged**